doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1178 @@
+1.  Optimized Bandwidth: By mapping VLANs to specific MSTIs, MSTP ensures that network paths are used efficiently, reducing link blocking and optimizing bandwidth usage
+ 
+#### MSTP Operation 
+
+MST BPDUs: MSTP uses Multiple Spanning Tree Bridge Protocol Data Units (MST BPDUs) to exchange information between switches. These BPDUs contain information about the MSTI and VLAN mappings, ensuring consistent spanning tree calculations across the network. 
+
+MSTP calculates spanning trees on the basis of Multiple Spanning Tree Bridge Protocol Data Units (MST BPDUs).
+
+<div align="center">
+<img src="images/MSTP_BPDU.png" alt="MSTP BPDU">
+<p>MSTP BPDU Format</p>
+</div><br>
+<div align="center">
+<img src="images/MSTI_config_message.png" alt="MSTP Config Message">
+<p>MSTI Configuration Messages</p>
+</div>
+
+*Refer to [RFC IEEE 802.1s-2002](https://standards.ieee.org/ieee/802.1s/1042/) for MSTP BPDU details.* 
+
+
+### Requirements 
+
+
+#### Functional Requirements 
+
+1. Support the creation of Multiple Spanning Tree Instances (MSTIs). 
+1. Support the assignment of one or more VLANs to a specific MSTI within a region. 
+
+1. Support the option to assign a region name and revision number to MSTP regions to achieve unique identification of VLAN to instance mapping across switches. 
+
+1. Support path selection and forwarding behavior in MSTI to optimize network performance within each instance by configuring a distinct root bridge. 
+
+1. Support the configuration of spanning tree parameters such as forward delay, hello timer, hop count. 
+
+1. The Destination Mac Address will be 01:80:C2:00:00:00 for MSTP BPDUs. 
+
+1. Support compatibility with networks employing different spanning tree protocols, such as STP, RSTP. 
+
+1. Support edge port functionality 
+
+1. Support BPDU guard functionality 
+
+1. Support root guard functionality 
+
+1. Support protocol operation on static breakout ports 
+
+1. Support protocol operation on Port-channel interfaces 
+
+ 
+#### Configuration & Management Requirements 
+
+
+1. Support CLI configurations as mentioned in Configuration section 
+
+1. Support show commands as mentioned in Configuration section 
+
+1. Support debug commands as mentioned in Configuration section 
+
+1. Support statistics commands as mentioned in Configuration section 
+
+1. Support clear commands as mentioned in Configuration section 
+
+1. Support disable commands as mentioned in Configuration section 
+
+# Architecture Design
+Following diagram explains the architectural design and linkages for MSTP. MSTP uses multiple existing SONiC containers, configuration details of each is mentioned below as well.
+
+![MSTP Architecture](images/MSTPDesign_Archi.drawio.png)
+## STP Container
+STP Container is responsible for actions taken for BPDU rx and BPDU tx. Following are the details for implementation:
```

Comment:
Please point me to the stp-container in sonic community code. I couldnt find it.
I see the following PR which is in review state:
https://github.com/sonic-net/sonic-buildimage/pull/3463

Reply:
SONIC has a repository as stp which holds all these details. Please see below link to repo:
https://github.com/sonic-net/sonic-stp

Reply:
Iam asking about the docker implementation in the PR https://github.com/sonic-net/sonic-buildimage/pull/3463

https://github.com/sonic-net/sonic-stp  -- provide stp implementation. I am looking for the stp docker, which is still in a PR https://github.com/sonic-net/sonic-buildimage/pull/3463

Reply:
Yes Stp docker is still in above open PR.

Reply:
If the code is still in a PR, how are we basing our design, which is not present in the code base yet?

Reply:
As discussed in the PENS meeting , we will be pushing the STP docker PR as well.